nifi-commons/nifi-security-utils/src/main/java/org/apache/nifi/security/util/KeyStoreUtils.java

```diff
@@ -245,7 +322,7 @@ public static TrustManagerFactory loadTrustManagerFactory(TlsConfiguration tlsCo
      */
     public static TrustManagerFactory loadTrustManagerFactory(String truststorePath, String truststorePassword, String truststoreType) throws TlsException {
         // Legacy truststore passwords can be empty
-        final char[] truststorePasswordChars = StringUtils.isNotBlank(truststorePassword) ? truststorePassword.toCharArray() : null;
+        final char[] truststorePasswordChars = StringUtils.isNotBlank(truststorePassword) ? truststorePassword.toCharArray() : "".toCharArray();
```

Comment:
@exceptionfactory I received a Null Pointer Exception for an empty password when the truststore type is PKCS12, so I changed it to an empty string. But after some investigation, I found that the Bouncy Castle PKCS12 store type does not allow empty passwords. 

Since we allow passwordless truststores, I'll add a check for the truststore type. If it's PKCS12, then I'll throw an Illegal Argument Exception, otherwise I'll set it back to `null`.

Reply:
@exceptionfactory After making these changes, I found out the real issue was that I received an NPE when loading a PKCS12 truststore and passing a `null` password. For loading a JKS or BCFKS store type, a `null` password is allowed. 

After our discussion and finding out the behavior among keystore/truststore types are different, we concluded to allow loading a store without a password, but not persisting a store without a password. These changes affect the methods I added to programmatically generate certificates, keystores, and truststores.